internal/cmd/commands/server/server.go

```diff
@@ -887,7 +894,7 @@ func (c *Command) reloadControllerDatabase(newConfig *config.Config) error {
 	}
 
 	// Acquire new lock on the new database and verify that it's in a good state to be used.
-	newDbSchemaManager, err := acquireDatabaseSharedLock(c.Context, newDb)
+	newDbSchemaManager, err := acquireSchemaManager(c.Context, newDb, c.Config.Controller.Database.SkipSharedLockAcquisition)
```

Comment:
```
if c.Config.Controller.Database.SkipSharedLockAcquisition {
			if err := c.schemaManager.Close(c.Context); err != nil {
				c.UI.Error(fmt.Errorf("Unable to release shared lock to the database: %w", err).Error())
				return base.CommandCliError
			}
		}
```

We need to do this for `newDbSchemaManager` too after we verify its state

Reply:
Good catch!